microshift_running_apps/microshift-greenboot.adoc

```diff
@@ -0,0 +1,88 @@
+
+Using greenboot can also reduce your risk of being locked out of an edge device during upgrades and prevent a significant interruption of service if an upgrade fails. When a failure is detected, the system boots into the last known working configuration using the `rpm-ostree` rollback capability.
+
+[NOTE]
+====
+The health-check procedures run on a system not booted using an OS-Tree file system, but no rollback is possible in the case of an upgrade failure.
+====
+
+include::modules/microshift-greenboot-install-rpm.adoc[leveloffset=+1]
+include::modules/microshift-greenboot-healthchecks-scripts.adoc[leveloffset=+1]
+
+The `40_microshift_running_check.sh` health check script is included in the `microshift-greenboot` RPM. The script validates that the required {product-title} services are running. Health check scripts are available in `/usr/lib/greenboot/check`, a read-only directory in RPM-OSTree systems.
+
+Exiting the script with a non-zero status will have the boot declared as failed. Greenboot redirects all the script output to the system log, accessible via the `journalctl -u greenboot-healthcheck.service` command.
+
+.Validation statuses and outcome for {product-title}.
+
+[cols="3", options="header"]
+|===
+|Validation
+|Pass
+|Fail
+
+|Check the script runs with 'root' permissions
+|Next
+|exit 0
+
+|Check microshift.service is enabled
+|Next
+|exit 0
+
+|Wait for microshift.service to be active (!failed)
+|Next
+|exit 1
+
+|Wait for Kubernetes API health endpoints to be OK
+|Next
+|exit 1
+
+|Wait for any Pod to start
+|Next
+|exit 1
+
+|For each core namespace, wait for images to be pulled
+|Next
+|exit 1
+
+|For each core namespace, wait for Pods to be ready
+|Next
+|exit 1
+
+|For each core namespace, check Pods not restarting
+|exit 0
+|exit 1
+|===
+
+The wait period in each validation is five minutes by default. After the wait period, if the validation has not succeeded, it is declared a failure. This wait period is incrementally increased by the base wait period after each boot in the verification loop. You can override the base time wait period by setting the MICROSHIFT_WAIT_TIMEOUT_SEC environment variable in the `/etc/greenboot/greenboot.conf` configuration file.
+
+[WARNING]
+====
+Wait until after an upgrade boot is declared successful before starting third-party workloads. If a rollback is performed after workloads start, you can lose data. Some third-party workloads create or update data on a device before an upgrade is complete. Upon rollback, the file system reverts to its state before the upgrade.
+====
+
+.Procedure
+
+* Run the following command to hold all workloads until an upgrade is declared valid:
```

Comment:
This command does not hold workloads. It merely checks when the boot success is declared.